File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        long goodPairs = 0;

        Map<Long, Long> count = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            long key = (long) nums[i] - i;

            long cnt = count.getOrDefault(key, 0L);

            goodPairs += cnt;

            count.put(key, cnt + 1);
        } 

        long n = nums.length;

        long totalPairs = (n * (n - 1)) / 2;

        return totalPairs - goodPairs;
    }
}

File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        stg = {}
        aPair = 0

        for i, num in enumerate(nums):
            key = num - i
            aPair += stg.get(key, 0)
            stg[key] = stg.get(key, 0) + 1

        n = len(nums)
        return (n * (n - 1)) // 2 - aPair